airflow/utils/dot_renderer.py

```diff
@@ -46,7 +47,9 @@ def _refine_color(color: str):
     return color
 
 
-def _draw_task(task: BaseOperator, parent_graph: graphviz.Digraph, states_by_task_id: Dict[str, str]) -> None:
+def _draw_task(
+    task: BaseOperator, parent_graph: graphviz.Digraph, states_by_task_id: Optional[Dict[Any, Any]]
```

Comment:
```suggestion
    task: BaseOperator, parent_graph: graphviz.Digraph, states_by_task_id: Dict[str, Any]
```

`states_by_task_id` should not be None; whatever is passing that in should be fixed instead.

Reply:
The function does support it being none:`if states_by_task_id:`

Reply:
Yep. That's the reason I changed it.